code/modules/research/nanites/nanite_programs/buffing.dm

```diff
@@ -64,15 +62,13 @@
 	. = ..()
 	if(ishuman(host_mob))
 		var/mob/living/carbon/human/H = host_mob
-		H.physiology.armor.laser += 30
-		H.physiology.armor.energy += 30
+		H.physiology.physio_armor.generate_new_with_specific(list(LASER = 20, ENERGY = 30))
 
 /datum/nanite_program/refractive/disable_passive_effect()
 	. = ..()
 	if(ishuman(host_mob))
 		var/mob/living/carbon/human/H = host_mob
-		H.physiology.armor.laser -= 30
-		H.physiology.armor.energy -= 30
+		H.physiology.physio_armor = null
```

Comment:
Same here, but this one also reduces the laser by 10 points.

Reply:
ditto